insta_reels_publishing_api_sample/index.js

```diff
@@ -0,0 +1,221 @@
+        res.redirect("/pages");
+    } catch (err) {
+        res.render("index", {
+            error: `There was an error with the request: ${err}`,
+        });
+    }
+});
+
+// Pages route to retrieve FB OAuth page tokens
+app.get("/pages", async function (req, res) {
+    const associatedPagesUri = `https://graph.facebook.com/v14.0/me/accounts?access_token=${req.session.userToken}`;
+    if (req.session.userToken) {
+        try {
+            const associatedPages = await axios.get(associatedPagesUri);
+            req.session.pageData = associatedPages.data.data;
+            res.render('upload_page', {
+                'pages': req.session.pageData
+            });
+        } catch (error) {
+            res.render("index", {
+                error: `There was an error with the request: ${error}`,
+            });
+        }
+    } else {
+        res.render("index", { error: "You need to log in first" });
+    }
+});
+
+app.post("/uploadReels", async function (req, res) {
+    const selectedPageID = req.body.pageID;
+    const pageToken = req.session.pageData.filter((pd) => pd.id === selectedPageID)[0].access_token;
+
+    try {
+         // Now Retrieve the Instgram user ID associated with the selected page
+        const getInstagramAccountUri = `https://graph.facebook.com/v14.0/${selectedPageID}?fields=instagram_business_account&access_token=${req.session.userToken}`;
+        const igResponse = await axios.get(getInstagramAccountUri);
+        const hasIgBusinessAccount = igResponse.data.instagram_business_account ? true : false;
+
+        // If there is a IG Business Account associated with the page
+        if(hasIgBusinessAccount) {
+            try {
+                const igUserId = igResponse.data.instagram_business_account.id;
+
+                // Upload Reel Video
+                const uploadVideoUri = `https://graph.facebook.com/v14.0/${igUserId}/media?media_type=REELS&video_url=${yourVideoUrl}&caption=${yourCaption}&access_token=${req.session.userToken}`;
+                const uploadResponse = await axios.post(uploadVideoUri);
+                const containerId = uploadResponse.data.id;
+
+                // add variables to the session
+                Object.assign(req.session, { igUserId, containerId, pageToken });
+
+                // Render Upload Success
+                res.render("upload_page", {
+                    uploaded: true,
+                    igUserId,
+                    containerId,
+                    pages: req.session.pageData,
+                    message: `Reel uploaded successfully on IG UserID #${igUserId} at Container ID #${containerId}. You can Publish now.`
+                });
+            } catch(e) {
+                res.render("upload_page", {
+                    uploaded: false,
+                    error: true,
+                    pages: req.session.pageData,
+                    message: `Error during upload. [Selected page id - ${selectedPageID}, IG User ID - ${igUserId}`
```

Comment:
if you ever get in to this catch block, `igUserId` will be undefined. So you need to move `igUserId` outside of the try block

Reply:
Good catch

Reply:
Done